mirdata/datasets/cipi.py

```diff
@@ -0,0 +1,259 @@
+        super().__init__(track_id, data_home, dataset_name, index, metadata)
+        self._data_home = data_home
+
+    @property
+    def title(self) -> str:
+        return (
+            self._track_metadata["work_name"]
+            if "work_name" in self._track_metadata
+            else None
+        )
+
+    @property
+    def book(self) -> str:
+        return self._track_metadata["book"] if "book" in self._track_metadata else None
+
+    @property
+    def URI(self) -> str:
+        return self._track_metadata["URI"] if "URI" in self._track_metadata else None
+
+    @property
+    def composer(self) -> str:
+        return (
+            self._track_metadata["composer"]
+            if "composer" in self._track_metadata
+            else None
+        )
+
+    @property
+    def musicxml_paths(self) -> List[str]:
+        return (
+            list(self._track_metadata["path"].values())
+            if "path" in self._track_metadata
+            else []
+        )
+
+    @property
+    def difficulty_annotation(self) -> str:
+        return (
+            self._track_metadata["henle"] if "henle" in self._track_metadata else None
+        )
+
+    def _check_embedding(self, fpath, file_type: str) -> str:
+        """
+        Verifies the existence of an embedding file and returns its path.
+
+        Args:
+            fpaths (str): The path to the embedding file.
+            file_type (str): The type of the embedding file.
+
+        Returns:
+            str: The path to the embedding file.
+
+        Raises:
+            FileNotFoundError: If the embedding file does not exist.
+        """
+        try:
+            with smart_open.open(fpath):
+                return fpath
+        except FileNotFoundError:
+            raise FileNotFoundError(
+                f"{file_type} embedding {fpath} for track {self.track_id} not found. "
+                "Did you run .download()?"
+            )
+
+    @core.cached_property
+    def fingering(self) -> tuple:
+        path_rh = self.get_path("rh_fingering")
+        path_lh = self.get_path("lh_fingering")
+        return self._check_embedding(path_rh, "Fingering"), self._check_embedding(
+            path_lh, "Fingering"
+        )
+
+    @core.cached_property
+    def expressiviness(self) -> str:
+        path = self.get_path("expressiviness")
+        return self._check_embedding(path, "Expressiviness")
+
+    @core.cached_property
+    def notes(self) -> str:
+        path = self.get_path("notes")
+        return self._check_embedding(path, "Expressiviness")
+
+    @core.cached_property
+    def scores(self) -> list:
+        try:
+            scores = [load_score(path, self._data_home) for path in self.musicxml_paths]
+        except FileNotFoundError:
+            raise FileNotFoundError(
+                "MusicXML file {} for track {} not found. "
+                "Did you run .download()?".format(self.musicxml_paths, self.track_id)
+            )
```

Comment:
I think this message is not well written! Because if a music xml file is not available, load_score will fail and specify which file is missing, but here, you will print the entire list... maybe you could just say something like "Missing MusicXML files. Did you run .download()?

Reply:
hey @PRamoneda sorry we missed that one. Can you take a look again?